apbs/pqr/reader.py

```diff
@@ -0,0 +1,85 @@
+from pyparsing import Group, Word, ZeroOrMore, alphas, alphanums, nums
+from apbs.chemistry import Atom, AtomList
+
+
+class PQRReader:
+    """A grammar/parser for PQR formatted data and files."""
+
+    def __init__(self):
+        """Define the grammar for an ATOM in PQR format."""
+        # TODO: Limit the field_name grammar to ATOM or HETOM literals
+        identifier = Word(alphas, alphanums + "_")
+        integer_val = Word(nums + "-")
+        float_val = Word(nums + "-" + ".")
+        atom_value = Group(
+            identifier("field_name")
+            + integer_val("atom_number")
+            + identifier("atom_name")
+            + identifier("residue_name")
+            + ZeroOrMore(identifier("chain_id"))
+            + integer_val("residue_number")
+            + ZeroOrMore(identifier("ins_code"))
+            + float_val("x")
+            + float_val("y")
+            + float_val("z")
+            + float_val("charge")
+            + float_val("radius")
+        )
+        self.atom = atom_value + ZeroOrMore(atom_value)
+
+    def loads(self, pqr_string: str) -> AtomList:
+        """
+        Find instances of atoms ignoring other syntax.
+
+        :param str pqr_string: One or more ATOM/HETOM
+        :return: the list of Atoms in the pqr_string
+        :rtype: AtomList
+        """
+        atoms = []
+        idx = 1
+        for item, _start, _stop in self.atom.scanString(pqr_string):
+            for match in item:
+                atom = Atom(
+                    field_name=match.field_name,
+                    atom_number=match.atom_number,
+                    atom_name=match.atom_name,
+                    residue_name=match.residue_name,
+                    chain_id=match.chain_id,
+                    residue_number=match.residue_number,
+                    ins_code=match.ins_code,
+                    x=match.x,
+                    y=match.y,
+                    z=match.z,
+                    charge=match.charge,
+                    radius=match.radius,
+                    id=idx,
+                )
```

Comment:
Suggestion:
```python
atom = Atom(**match)
```

Reply:
@sobolevnrm do you prefer being verbosely explicit or brief with @ashermancinelli suggestion?

Reply:
Explicit unless it is too painful.

Reply:
I think for a grammar it is better to be explicit and I don't know the threshold for "pain".